src/mbgl/annotation/annotation_manager.cpp

```diff
@@ -167,13 +159,26 @@ void AnnotationManager::updateStyle(Style& style) {
         shape.second->updateStyle(style);
     }
 
+    for (const auto& image : images) {
+        if (!style.getImage(image.first)) {
+            style.addImage(image.first, std::make_unique<style::Image>(image.second));
```

Comment:
Can we move images instead of copying them? This will leave the Image object in the `unordered_map` empty, but we'll never access it again (there's no `getImage` method on the `AnnotationManager`). We can treat the mere presence of the key as the availability of this image, and only call `style.addImage` when we still own the image.